src/masternodes/vaulthistory.h

```diff
@@ -55,14 +55,62 @@ struct VaultHistoryValue {
     }
 };
 
+struct VaultStateKey {
+    uint256 vaultID;
+    uint32_t blockHeight;
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action) {
+        READWRITE(vaultID);
+
+        if (ser_action.ForRead()) {
+            READWRITE(WrapBigEndian(blockHeight));
+            blockHeight = ~blockHeight;
+        }
+        else {
+            uint32_t blockHeight_ = ~blockHeight;
+            READWRITE(WrapBigEndian(blockHeight_));
+        }
+    }
+};
+
+struct VaultStateValue {
+    bool isUnderLiquidation;
+    TAmounts collaterals;
+    CCollateralLoans collateralsValues;
+    std::vector<CAuctionBatch> auctionBatches;
+    uint32_t ratio;
```

Comment:
We can remove `ratio`, `collateralsValues` provides it by calling its method.

Reply:
We want the ratio and collateralsValues stored from that point in time.

Reply:
Correct, `collateralsValues.ratio()` is the same as `ratio` no?

Reply:
I understand now, I misunderstood. This does not work, the state displayed by getvaulthistory is not really the actual state, it is a composite of state from the start and end of ProcessLoanEvents. The ratio shown is from before liquidation, where CCollateralLoans is stored after liquidation where the collateral and loans have been moved out of CCollateralLoans and into auction batches.